indico/modules/rb_new/client/js/components/modals/BookingDetailsModal.jsx

```diff
@@ -0,0 +1,181 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import _ from 'lodash';
+import moment from 'moment';
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import {Button, Modal, Message, Grid, Header, ModalActions, Icon, Popup} from 'semantic-ui-react';
+import {Param, Translate} from 'indico/react/i18n';
+import {toMoment, serializeDate} from 'indico/utils/date';
+import BookingTimeInformation from '../BookingTimeInformation';
+import RoomBasicDetails from '../../components/RoomBasicDetails';
+import {selectors as bookingsSelectors} from '../../common/bookings';
+import TimelineContent from '../../components/TimelineContent';
+import TimelineLegend from '../../components/TimelineLegend';
+
+import './BookingDetailsModal.module.scss';
+
+
+class BookingDetailsModal extends React.Component {
+    static propTypes = {
+        booking: PropTypes.object.isRequired,
+        onClose: PropTypes.func,
+    };
+
+    static defaultProps = {
+        onClose: () => {}
```

Comment:
i think we usually assign it to `null` and then check whether it is not nullish when trying to call

Reply:
I think having a callable default is much better :)